internal/controller/controller.go

```diff
@@ -59,14 +59,21 @@ const (
 	// refresh its cache.
 	DefaultResyncPeriod = 10 * time.Minute
 
-	// DefaultReconciliationPeriod is the period after which all the objects in
-	// the informer's cache will be added to the workqueue so that they can be
+	// reconciliationPeriod is the period after which all the objects in the
+	// informer's cache will be added to the workqueue so that they can be
 	// processed by the syncHandler().
 	//
 	// The informer calls the event handlers only if the resource state
-	// changes. We force this additional reconciliation as a liveness check to
-	// see if the operator is working as intended.
-	DefaultReconciliationPeriod = 5 * time.Minute
+	// changes. We do this additional reconciliation as a liveness check to see
+	// if the operator is working as intended.
+	reconciliationPeriod = 5 * time.Minute
+
+	// maintenancePeriod is the period after which the worker will clean up any
+	// orphaned absent alerts across the entire cluster.
+	//
+	// We do this manual cleanup in case a PrometheusRule is deleted and the
+	// update of the shared informer's cache has missed it.
+	maintenancePeriod = 6 * time.Hour
```

Comment:
Let's make this quite a bit shorter, e.g. 1 hour. It shouldn't hurt much to do it more often since we usually just list all promrules once and are done with it.